tests/raftstore/transport_simulate.rs

```diff
@@ -359,9 +373,9 @@ impl PauseFirstSnapshotFilter {
 }
 
 impl Filter<StoreMsg> for PauseFirstSnapshotFilter {
-    fn before(&self, msgs: &mut Vec<StoreMsg>) {
+    fn before(&self, msgs: &mut Vec<StoreMsg>) -> Result<()> {
         if self.stale.load(Ordering::Relaxed) {
-            return;
+            return check_messages(msgs);
```

Comment:
check_messages() will return `Ok(())` if msgs is not empty. An error returned by `before` will be delivered to `SimulateTransport`.